rasa/core/actions/forms.py

```diff
@@ -593,10 +616,21 @@ async def activate(
             return []
 
         logger.debug(f"Validating pre-filled required slots: {prefilled_slots}")
-        return await self.validate_slots(
+
+        validate_events = await self.validate_slots(
             prefilled_slots, tracker, domain, output_channel, nlg
         )
 
+        validated_slot_names = [
+            event.key for event in validate_events if isinstance(event, SlotSet)
+        ]
+
+        return validate_events + [
+            event
+            for event in extraction_events
+            if isinstance(event, SlotSet) and event.key not in validated_slot_names
+        ]
```

Comment:
would be rad to be able to do compare events with `!=` or `==` 🤔 I thought this was possible?

Reply:
🤔 I see that `__eq__` raises `NotImplemented`, which is really not helpful ahah

Reply:
I think implementing `__eq__` in events is out of scope here 😬 Happy to submit a ticket though for a clean-up?

Reply:
yeah of course. I am not sure how much it's worth it and how much it would simplify the code, maybe that's worth having a technical spike for that 🤔

Reply:
Opened a [spike ticket](https://rasahq.atlassian.net/browse/ATO-128)